fbw-a32nx/src/systems/fmgc/src/guidance/Geometry.ts

```diff
@@ -324,6 +328,19 @@ export class Geometry {
         leg.recomputeWithParameters(activeLegIdx === index, legPredictedTas, legPredictedGs, ppos, trueTrack);
       }
     }
+
+    const element = plan.legElementAt(index);
```

Comment:
The geometry leg should also contain a reference to `calculated`. Are you sure you need to get the flight plan leg here?

Reply:
Hmm, having this in both places without a clear constraint that they are the same object is sus. Let me think about this more.